marine.c

```diff
@@ -966,6 +980,363 @@ WS_DLL_PUBLIC void marine_free(marine_result *ptr) {
     }
 }
 
+static marine_packet_field *
+marine_packet_field_init(marine_packet_field *node, char *name) {
```

Comment:
This can be `void`, you don't need to return the struct (and indeed you're not using the return value anywhere)